### PR TITLE
enable test for setting custom header

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/autorest.go/issues/268

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/autorest.go/issues/298

File: src/generator/helpers.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { ArraySchema, CodeModel, DictionarySchema, Language, Parameter, Schema,
 import { ImportManager } from './imports';
 
 export const dateFormat = '2006-01-02';
-export const datetimeRFC3339Format = 'time.RFC3339';
+export const datetimeRFC3339Format = 'time.RFC3339Nano';
```
Comment (catalinaperalta) [inline src/generator/helpers.ts:14]: Just wondering, there no case where we need time.RFC3339?

Comment (catalinaperalta) [reply src/generator/helpers.ts:14]: Or do the trailing zeros not matter?

Comment (jhendrixMSFT) [reply src/generator/helpers.ts:14]: Omitting the trailing zeros just saves a few bytes, which in itself isn't a big deal, however we use Nano in other places so it's more about consistency.

Comment (catalinaperalta) [reply src/generator/helpers.ts:14]: Ok, got it 👍

**Pull request merged**